libs/platform/ebpf_epoch.c

```diff
@@ -520,13 +545,22 @@ _ebpf_epoch_get_release_epoch(_Out_ int64_t* release_epoch)
                     ebpf_queue_non_preemptible_work_item(_ebpf_epoch_cpu_table[cpu_id].stale_worker, NULL);
                 }
             } else {
-                _ebpf_set_per_cpu_flag(&_ebpf_epoch_cpu_table[cpu_id], EBPF_EPOCH_PER_CPU_STALE, true);
+                _ebpf_epoch_cpu_table[cpu_id].epoch_state.stale = true;
+            }
+            if (!_ebpf_epoch_cpu_table[cpu_id].epoch_state.timer_armed &&
+                !_ebpf_epoch_cpu_table[cpu_id].epoch_state.shutdown) {
+
+                _ebpf_epoch_cpu_table[cpu_id].epoch_state.timer_armed = true;
+
```

Comment:
suggestion: keeping `stale_items` local variable and scheduling the timer at the end of the function might reduce the code delta a bit.